### ISSUE TITLE
[Decision] Sequencer Deployment Options

### ISSUE THREAD

**Issue by RossiNYC** [pokt-network/poktroll#32]
## Objective

Verify options for the Shannon sequencer deployment and present to PNF/Community for decision. 

## Origin Document

TL;DR: PNF/DAO needs to decide who/how the Shannon Sequencer will be deployed AND the process required for making the decision (e.g. Community Vote)
See: https://www.notion.so/poktdao/Sequencer-Options-Shannon-bc11a37f685343188dd12b1a86fe459b for more context. 

SKIP Cost Estimation: https://skip-protocol.notion.site/Skip-Ecosystem-Development-Kit-EDK-Pricing-for-POKT-6f9598a3ff0c411bbf8d54830a520ef8 (Proposal withdrawn - no longer an option)

Caldera RFP Response: https://docs.google.com/document/d/1P3WOGzVB92kR6VgS3GsBId-uzHI2URJM7AC6BGRgV7U/edit

## Goals

Verify options (e.g. Does Rollkit/Celestia support CosmoSDK x/Gov) 
If outsourcing to a 3rd party - Identify 3rd Parties under consideration (e.g. SKIP) 

## Deliverables

- [ ] Clear documentation on the process of making the Sequencer decision
- [ ] Un-bias explainer document for all vetted options including decision matrix/factors (e.g. Cost, Time to Deploy, etc.)
- [ ] Risks/Mitigations for each option
- [ ] Recommendation Document for DAO voting 

## Non-goals / Non-deliverables

The actual decision, this Issue lays out what is needed to get to the decision, not the decision itself. 
The implementation details of the final decision

**Creator**: @RossiNYC 
**Co-Owners**: Ben @ PNF

**Comment by RossiNYC** [comment]
PNF has reviewed the SKIP cost documentation https://skip-protocol.notion.site/Skip-Ecosystem-Development-Kit-EDK-Pricing-for-POKT-6f9598a3ff0c411bbf8d54830a520ef8 - B3n has been added to the TG chat and we're trying to organize a call for this week. Meanwhile, reviewing other options for comparison/de-risking.

**Comment by RossiNYC** [comment]
Additionally, I looked at “shared sequencer” options that are part of the Celestia ecosystem: 
Astria:  https://www.astria.org/ - looks like they have a testnet ready to go - they are the furthest ahead from what I can gather from websites
Radius: https://www.theradius.xyz/ - docs are there but not much else, they recently raised, looks like a work in progress 
Fairblock: https://www.fairblock.network/ also just raised 2.5M in Oct, I like the idea a lot but they are probably 12-18 months out from a testnet.

**Comment by Olshansk** [comment]
Posting a quick update:
1. The team at [skip.money](https://skip.money/) withdrew their proposal (see screenshot below).
2. The team at [caldera](https://caldera.xyz/) is already experimenting with rollkit for other chains (see screenshot below).

My recommendations for next steps are:
1. Leverage the support from Skip to have a meeting and collect requirements of what we need.
2. Meet with the team from Caldera to understand the overlap/differences of their offerings and the requirements we collect based on (1).
3. Use the results of (1) & (2) to formalize our requirements, get an estimate of costs and then the following:
- Put together a proposal (via the DAO) to let the community (i.e. Pocket validator runners) bid on the project
- Enable other RaaS providers bid on the project alongside our community.

@RossiNYC Wanted to know what you think and if you'd be able to lead the charge on this?

---

![Skip](https://github.com/pokt-network/poktroll/assets/1892194/46ce8582-0642-462e-9317-58c7443fe7b0)

![Caldera](https://github.com/pokt-network/poktroll/assets/1892194/52a1aad2-d08f-42b6-b322-8df8d4c8833b)

**Comment by Olshansk** [comment]
@RossiNYC Please post our sequencer requirements here once we have them ironed out. Thank you!

**Comment by Olshansk** [comment]
Posting @jacklaing's notes after meeting with the Espresso Systems team:

```
- Decentralized sequencer uses hotstuff, reaches finality after 3 blocks, and scales to any block time - won’t be a bottleneck.
- They have a DA layer called tiramisu that runs in parallel to consensus, but teams can use other DA like Celestia.
- They’ve been giving Celestia feedback on Rollkit. Sequencers in Celestia are DA-aware because they’re required to pay fees to DA providers but Espresso’s sequencer intentionally holds minimal state so doesn’t do fee tracking for what the sequencer needs to pay third-parties (only minimal fee tracking of what is needed to cover Espresso’s own costs).
- Decentralized Sequencer Roadmap: smaller testnet December (2 node operators), larger testnet January, mainnet by end of Q1.
- Will be doing a code freeze followed by inter-team code audit (they have 3 teams - sequencer, consensus, and cryptography) and an external audit.
Partners: Injective, Caldera, Altlayer, Lumos, Off-chain Labs
- Haven’t landed on pricing yet - having pricing discovery calls with partners
- Support: shared Slack channels with partners or Telegram groups, whichever the partner prefers.
- Docs are kind of barebones at the moment but will improve in the next few weeks as we get closer to testnet.
- Other partner opportunities: seemed interested in Pocket being used by their partners to send transactions to the sequencer (via RPC), subject to partner interest
```

**Comment by Olshansk** [comment]
Sharing the RFP @RossiNYC put together [here](https://docs.google.com/document/d/1aPyVW2grQP-f-bvF7MmjOZI2kuXclDY7d0IjBkmkmWk).

**Comment by Olshansk** [comment]
Summarizing some notes from a [conversation](https://www.notion.so/buildwithgrove/Skip-Pocket-Cross-Team-Meetings-1a121cc25ea54a56a1e0af31133782f4?pvs=4) I had with some of our advisors on this matter:
- [Astria](https://www.astria.org/)
	- Less mature but very promising
	- Likely to provide a lot more hands-on support at a good price
- [Caldera](https://caldera.xyz/)
	- Heavily EVM focused but experimenting with Rollkit
	- Will provide some hands-on support
- [Conduit](https://conduit.xyz/)
	- Heavily focused on the EVM
	- Might not be able to provide a lot of hands-on support
- [Strangelove](https://strange.love/)
	- 	Cosmos native team that is very skilled and very experienced but likely to be very expensive
- [Binary](https://binary.builders/)
	- 	Cosmos native team that would be open to helping us

**Comment by RossiNYC** [comment]
Caldera RFP Response: https://docs.google.com/document/d/1P3WOGzVB92kR6VgS3GsBId-uzHI2URJM7AC6BGRgV7U/edit 

I also updated the notion page https://www.notion.so/poktdao/Sequencer-Options-Shannon-bc11a37f685343188dd12b1a86fe459b with their answers and included the requirements table we sent out with the RFP so we can line up the responses vs. req.

**Comment by Olshansk** [comment]
Closing this out as the decision is to:

1. Go with Astria
2. Use Caldera or run our own if timelines misalign.
